src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -462,9 +460,7 @@ export class DataSourceSavedObjectsClientWrapper {
   }
 
   private async encryptSigV4Credential<T = unknown>(auth: T, encryptionContext: EncryptionContext) {
-    const {
-      credentials: { accessKey, secretKey, region, service },
-    } = auth;
+    const { accessKey, secretKey, region, service } = auth as SigV4Content;
```

Comment:
Same question as Miki that we should destruct auth.credentials to get all the credentials related fields.

Reply:
Many thanks for pointing this out. would the following work?

`    const {
      credentials: { accessKey, secretKey, region, service },
    } = auth as { credentials: SigV4Content };`

Reply:
Many thanks for pointing this out. It was my mistake and got confused about the structure. Would this work?

`    const {
      credentials: { accessKey, secretKey, region, service },
    } = auth as { credentials: SigV4Content };
`